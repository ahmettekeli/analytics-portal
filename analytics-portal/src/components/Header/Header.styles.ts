import styled from "styled-components";
//TODO Header styling.
