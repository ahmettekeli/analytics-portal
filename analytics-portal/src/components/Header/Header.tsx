import { StyledHeader } from "./Header.styles";
import Breadcrumbs from "../Breadcrumbs/Breadcrumbs";

function Header() {
  return (
    <StyledHeader data-testid="Header">
      <Breadcrumbs />
    </StyledHeader>
  );
}

export default Header;
