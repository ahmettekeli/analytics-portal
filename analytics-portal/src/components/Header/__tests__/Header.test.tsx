import { cleanup, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import Header from "../Header";
import { BrowserRouter as Router } from "react-router-dom";

describe("Header", () => {
  afterEach(cleanup);

  test("Renders Header component", () => {
    const renderResult = render(
      <Router>
        <Header />
      </Router>
    );
    expect(renderResult.getByTestId("Header")).toBeInTheDocument();
  });

  test("Header matches snapshot", () => {
    const tree = renderer
      .create(
        <Router>
          <Header />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
