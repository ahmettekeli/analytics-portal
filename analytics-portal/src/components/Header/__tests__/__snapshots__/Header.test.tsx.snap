// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header matches snapshot 1`] = `
<div
  className="sc-bdvvtL kjOdND"
  data-testid="Header"
>
  <nav
    aria-label="breadcrumb"
    className="MuiTypography-root MuiBreadcrumbs-root sc-pVTFL jDjMEW MuiTypography-body1 MuiTypography-colorTextSecondary"
    data-testid="Breadcrumbs"
  >
    <ol
      className="MuiBreadcrumbs-ol"
    >
      <li
        className="MuiBreadcrumbs-li"
      >
        <a
          className="sc-furwcr iPgPam"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
    </ol>
  </nav>
</div>
`;
